### ISSUE TITLE
Errors when trying to run tests

### ISSUE THREAD

**Issue by yogodoshi** [sorentwo/oban#6]
Hey 👋 ! I'm trying to switch from exq to oban but I'm having issues with the tests. I've followed the [recommendations on the README related to testing](https://github.com/sorentwo/oban/tree/v0.2.0#testing) but when I run my tests I get a lot of errors that seem to be related to processes that Oban creates in the background:

```
13:32:18.928 [error] GenServer Oban.Queue.Default.Producer terminating
** (DBConnection.OwnershipError) cannot find ownership process for #PID<0.429.0>.

When using ownership, you must manage connections in one
of the four ways:

* By explicitly checking out a connection
* By explicitly allowing a spawned process
* By running the pool in shared mode
* By using :caller option with allowed process

The first two options require every new process to explicitly
check a connection out or be allowed by calling checkout or
allow respectively.

The third option requires a {:shared, pid} mode to be set.
If using shared mode in tests, make sure your tests are not
async.

The fourth option requires [caller: pid] to be used when
checking out a connection from the pool. The caller process
should already be allowed on a connection.

If you are reading this error, it means you have not done one
of the steps above or that the owner process has crashed.

See Ecto.Adapters.SQL.Sandbox docs for more information.
    (ecto_sql) lib/ecto/adapters/sql.ex:615: Ecto.Adapters.SQL.raise_sql_call_error/1
    (ecto_sql) lib/ecto/adapters/sql.ex:551: Ecto.Adapters.SQL.execute/5
    (ecto) lib/ecto/repo/queryable.ex:153: Ecto.Repo.Queryable.execute/4
    (oban) lib/oban/queue/producer.ex:180: Oban.Queue.Producer.fetch_jobs/3
    (oban) lib/oban/queue/producer.ex:170: Oban.Queue.Producer.dispatch/1
    (stdlib) gen_server.erl:637: :gen_server.try_dispatch/4
    (stdlib) gen_server.erl:388: :gen_server.loop/7
    (stdlib) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Last message: {:continue, :start}
13:32:18.951 [error] GenServer Oban.Queue.Default.Watchman terminating
** (stop) exited in: GenServer.call(Oban.Queue.Default.Producer, :pause, 5000)
    ** (EXIT) no process: the process is not alive or there's no process currently associated with the given name, possibly because its application isn't started
    (elixir) lib/gen_server.ex:979: GenServer.call/3
    (oban) lib/oban/queue/watchman.ex:43: Oban.Queue.Watchman.terminate/2
    (stdlib) gen_server.erl:673: :gen_server.try_terminate/3
    (stdlib) gen_server.erl:858: :gen_server.terminate/10
    (stdlib) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Last message: {:EXIT, #PID<0.426.0>, :shutdown}
13:32:18.952 [error] GenServer Oban.Queue.Default.Producer terminating
** (DBConnection.OwnershipError) cannot find ownership process for #PID<0.453.0>.

When using ownership, you must manage connections in one
of the four ways:

* By explicitly checking out a connection
* By explicitly allowing a spawned process
* By running the pool in shared mode
* By using :caller option with allowed process

The first two options require every new process to explicitly
check a connection out or be allowed by calling checkout or
allow respectively.

The third option requires a {:shared, pid} mode to be set.
If using shared mode in tests, make sure your tests are not
async.

The fourth option requires [caller: pid] to be used when
checking out a connection from the pool. The caller process
should already be allowed on a connection.

If you are reading this error, it means you have not done one
of the steps above or that the owner process has crashed.

See Ecto.Adapters.SQL.Sandbox docs for more information.
    (ecto_sql) lib/ecto/adapters/sql.ex:615: Ecto.Adapters.SQL.raise_sql_call_error/1
    (ecto_sql) lib/ecto/adapters/sql.ex:551: Ecto.Adapters.SQL.execute/5
    (oban) lib/oban/queue/producer.ex:116: Oban.Queue.Producer.rescue_orphans/1
    (oban) lib/oban/queue/producer.ex:49: Oban.Queue.Producer.handle_continue/2
    (stdlib) gen_server.erl:637: :gen_server.try_dispatch/4
    (stdlib) gen_server.erl:388: :gen_server.loop/7
    (stdlib) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Last message: {:continue, :start}
... (this happens more ~15 times)
```

I've tried a few different strategies using `Ecto.Adapters.SQL.Sandbox.allow` and different Sandbox `:shared` configs but I couldn't stop those errors as those processes seems to be created when the app starts, even before the tests start to run, right?

Here are the parts my configs that might be helpful:
```
# test_helper.exs
...
ExUnit.start()
Ecto.Adapters.SQL.Sandbox.mode(Pidgey.Repo, :manual)


# application.ex
def start(_type, _args) do
    children = [
      Pidgey.Repo,
      PidgeyWeb.Endpoint,
      Pidgey.Vault,
      {Oban, Application.get_env(:pidgey, Oban)}
    ]

    opts = [strategy: :one_for_one, name: Pidgey.Supervisor]

    Supervisor.start_link(children, opts)
  end

# config.exs
config :pidgey, Oban,
  repo: Pidgey.Repo,
  queues: [default: 10]

# config/test.exs
config :pidgey, Oban, poll_interval: :timer.minutes(60)
```

It is a Phoenix app, some tests are using `ExUnit.Case`, others `DataCase` or `ConnCase` but when trying to run even a single test file those errors will be raised, even though the specs will pass.

I'm creating this issue because I think it might be an issue with the package, let me know if it is not and that I should create a question on StackOverflow or on the Elixir forum :)

**Comment by sorentwo** [comment]
@yogodoshi Thank you for the report! You're absolutely right, this was an issue with the package (or at least how it was instructing you to configure it). When the testing docs were written there wasn't any background connection activity with a lengthy polling time. That has changed and the docs needed to be updated accordingly.

In addition to doc updates in 629444b I've also made it easier to disable queues in the test environment. Set `queues: false` and `prune: :disabled` within your `config/test.exs` file:

```elixir
config :pidgey, Oban, queues: false, prune: :disabled
```

**Comment by sorentwo** [comment]
Note that this fix will be included in `0.3`, which I aim to release this week. In the meantime please give master a try and let me know if it fixes things for you. 👍

**Comment by yogodoshi** [comment]
Switched to use master, followed the updated test guideline and now the errors are gone. Thanks for the help ❤️
